src/Arcus.Messaging.Tests.Runtimes.AzureFunction.EventHubs.InProcess/Startup.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using Arcus.EventGrid.Publishing;
+using Arcus.Messaging.Tests.Core.Messages.v1;
+using Arcus.Messaging.Tests.Runtimes.AzureFunction.EventHubs.InProcess;
+using Arcus.Messaging.Tests.Workers.MessageHandlers;
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
+
+[assembly: FunctionsStartup(typeof(Startup))]
+
+namespace Arcus.Messaging.Tests.Runtimes.AzureFunction.EventHubs.InProcess
+{
+    public class Startup : FunctionsStartup
+    {
+        /// <summary>
+        /// This method gets called by the runtime. Use this method to add services to the container.
+        /// </summary>
+        /// <param name="builder">The instance to build the registered services inside the functions app.</param>
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            builder.Services.AddTransient(serviceProvider =>
+            {
+                var eventGridTopic = Environment.GetEnvironmentVariable("EVENTGRID_TOPIC_URI");
+                var eventGridKey = Environment.GetEnvironmentVariable("EVENTGRID_AUTH_KEY");
+
+                return EventGridPublisherBuilder
+                       .ForTopic(eventGridTopic)
+                       .UsingAuthenticationKey(eventGridKey)
+                       .Build();
+            });
+            
+            builder.AddEventHubsMessageRouting()
+                   .WithEventHubsMessageHandler<OrderEventHubsMessageHandler, Order>();
+        }
```

Comment:
Since this PR is to simplify correlation support, I can see in some method here that the TelemetryClient is being retrieved from the services container.   However, I would suppose that you need to somehow register the TelemetryClient here ?  Or how is it done ?

Reply:
That was already done in another PR: https://github.com/arcus-azure/arcus.messaging/blob/6b17fdd5b630c1918dceb2b0f4cc337e76d28ff1/src/Arcus.Messaging.Abstractions/Extensions/IServiceCollectionExtensions.cs#L67
The simplification here is to not have a direct reference to the `TelemetryClient`, like the PR's description.